src/layouts/govuk.njk

```diff
@@ -14,17 +14,26 @@
 {% endblock %}
 
 {% block head %}
+  <meta name="csrf-token" content="{{context.csrf}}" />
+  <meta name="x-user-identity-origin" content="{{context.origin}}" />
   <!-- Global site tag (gtag.js) - Google Analytics -->
   <script async src="https://www.googletagmanager.com/gtag/js?id=UA-43115970-5"></script>
+
+  <!-- Pull dimensions vals from meta ; else all script/origin combinations have to be in the CSP -->
   <script>
     window.dataLayer = window.dataLayer || [];
     function gtag(){dataLayer.push(arguments);}
     gtag('js', new Date());
 
-    gtag('config', 'UA-43115970-5');
-  </script>
+    var dimensions = {};
 
-  <meta name="csrf-token" content="{{context.csrf.toString()}}" />
+    var originTag = document.querySelector('meta[name="x-user-identity-origin"]')
+    if (originTag && originTag.content) {
+      dimensions['dimension1'] = originTag.content;
+    }
+
+    gtag('config', 'UA-43115970-5', dimensions);
```

Comment:
I'm not sure this is right according to the google docs [you linked](https://developers.google.com/analytics/devguides/collection/gtagjs/custom-dims-mets):

```js
// Configures custom dimension<Index> to use the custom parameter
// 'dimension_name' for 'GA_MEASUREMENT_ID', where <Index> is a number
// representing the index of the custom dimension.
gtag('config', 'GA_MEASUREMENT_ID', {
  'custom_map': {'dimension<Index>': 'dimension_name'}
});

// Sends the custom dimension to Google Analytics.
gtag('event', 'any_event_name', {'dimension_name': dimension_value});
```

Shouldn't it be:

```js
gtag('config', 'UA-43115970-5', {
  'custom_map': {'dimension1': 'origin'}
});
gtag('event', 'origin_dimension', {'origin': originTag.content});
```

Or something? I'm not 100% sure - the docs are quite confusing.

Reply:
After reviewing the page I linked I realised I linked a not-helpful page.

`custom_map` is renaming things in the console, along with the implicit event.

My understanding is calling `gtag` with `'config'` as the first argument does two things:

- configure `gtag` with the ID (`UA-*`) and with any additional properties
- implicitly send a pageview event

So calling `'config'` and then calling `'event'` AFAICT is two events, but we only want one.

I think what we want is:

```
gtag('set', 'dimension1', originTag.content);
gtag('config', 'UA-43115970-5')
```

which is longhand for

```
gtag('config', 'UA-43115970-5', dimensions);
```

---

Info about `'set'`, `'event'`, and `'config'` [here](https://developers.google.com/gtagjs/reference/api)

Gohugo.io uses Google analytics in this fashion, search for `dimension1` within `view-source:https://gohugo.io/`